exploration/name-shadowing.md

```diff
@@ -33,23 +33,31 @@ The term **local variable** refers to a value that is defined in a declaration.
 The term **external variable** refers to a value that is passed to the
 message formatter by name and which can be referred to in an expression.
```

Comment:
Looking at the changes you've proposed, I think this part ought to be revisited as well. Given that the chosen solution is to use `input` and `local` declarations, the definitions here of "local" and "external" variables don't necessarily align all that well. Maybe we'd be better off with "local", "input", _and_ "implicit" variables as separate concepts?

Overall, though, I'm not excessively keen on spending too much time on this, esp. given that the corresponding changes have already been applied to the spec. If there are issues with any of that text, that should definitely be prioritised. The remaining value of this doc is to provide some explanation of our decisions for later really interested readers.